tests/utils_tests/test_simplelazyobject.py

```diff
@@ -29,3 +35,16 @@ def test_pickle_py2_regression(self):
 
             # This would fail with "TypeError: expected string or Unicode object, NoneType found".
             cPickle.dumps(x)
+
+    def test_pickle_model_instance_with_one_to_one_field(self):
+
+        # See ticket #25426
+        user = User.objects.create_user('johndoe', 'john@example.com', 'pass')
+        UserProfile.objects.create(user=user)
+
+        x = SimpleLazyObject(lambda: user)
+        x.profile  # accsss related object
+
+        # This would fail with "RuntimeError: dictionary changed size during iteration",
+        # on django 1.8 both on python3.x and 2.x
+        pickle.dumps(x)  # Fails
```

Comment:
This should also unpickle the model, and make sure there is no warning.  Or I can add that in #5276.